src/main/kotlin/com/zepben/auth/client/ZepbenTokenFetcher.kt

```diff
@@ -36,76 +35,57 @@ import java.time.Instant
  * Fetches access tokens from an authentication provider using the OAuth 2.0 protocol.
  *
  * @property audience Audience to use when requesting tokens.
- * @property issuerDomain The domain of the token issuer.
+ * @property tokenEndpoint The token endpoint of the issuer; used to fetch the token.
  * @property authMethod The authentication method used by the server.
- * @property issuerProtocol Protocol of the token issuer. You should not change this unless you are absolutely sure of
- *                          what you are doing. Setting it to anything other than https is a major security risk as
- *                          tokens will be sent in the clear.
- * @property requestContentType  content type for the OAUTH2 request.
- * @property tokenPath Path for requesting token from `issuer_domain`.
  * @property tokenRequestData Data to pass in token requests.
  * @property refreshRequestData Data to pass in refresh token requests.
  * @property client HTTP client used to retrieve tokens. Defaults to HttpClient.newHttpClient().
  * @property refreshToken Refresh Token; will be used if defined (fetched previously).
- * @property createBody a callback to turn the <*>RequestData into a string. AUTH0 requires JSON representation, where AZURE is query params.
+ * @property requestContentType  content type for the OAUTH2 request.
+ * @property createBody a callback to turn the <*>RequestData into a request body. AUTH0 requires JSON representation, where EntraID (Azure) is query params.
  */
 class ZepbenTokenFetcher(
     val audience: String,
-    val issuerDomain: String,
-    val authMethod: AuthMethod,
-    val issuerProtocol: String = "https",
-    tokenPath: String? = null,
+    val tokenEndpoint: String,
+    val authMethod: AuthMethod = AuthMethod.OAUTH,
     val tokenRequestData: JsonObject = JsonObject(),
     val refreshRequestData: JsonObject = JsonObject(),
     private val client: HttpClient = HttpClient.newHttpClient(),
     private var refreshToken: String? = null,
-    val requestContentType: String = "application/json",
+    val requestContentType: String = "application/x-www-form-urlencoded",
```

Comment:
Forgot to mention, since there is only ever one content type, we can delete all refs to this right?